conans/client/tools/apple.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import subprocess
+
+
+def cmd_output(command):
+    return subprocess.check_output(command).decode().strip()
+
+
+def is_apple_os(_os):
+    """returns True if OS is Apple one (Macos, iOS, watchOS or tvOS"""
+    return str(_os) in ['Macos', 'iOS', 'watchOS', 'tvOS']
+
+
+def to_apple_arch(arch):
+    """converts conan-style architecture into Apple-style arch"""
+    return {'x86': 'i386',
+            'x86_64': 'x86_64',
+            'armv7': 'armv7',
+            'armv8': 'arm64',
+            'armv7s': 'armv7s',
+            'armv7k': 'armv7k'}.get(str(arch))
+
+
+def apple_sdk_name(settings):
+    """returns proper SDK name suitable for OS and architecture
+    we're building for (considering simulators)"""
+    arch = settings.get_safe('arch')
+    _os = settings.get_safe('os')
+    if str(arch).startswith('x86'):
+        return {'Macos': 'macosx',
+                'iOS': 'iphonesimulator',
+                'watchOS': 'watchsimulator',
+                'tvOS': 'appletvsimulator'}.get(str(_os))
+    elif str(arch).startswith('arm'):
+        return {'iOS': 'iphoneos',
+                'watchOS': 'watchos',
+                'tvOS': 'appletvos'}.get(str(_os))
+    else:
+        return None
+
+
+def apple_deployment_target_env_name(_os):
+    """environment variable name which controls deployment target"""
+    return {'Macos': 'MACOSX_DEPLOYMENT_TARGET',
+            'iOS': 'IOS_DEPLOYMENT_TARGET',
+            'watchOS': 'WATCHOS_DEPLOYMENT_TARGET',
+            'tvOS': 'TVOS_DEPLOYMENT_TARGET'}.get(str(_os))
+
+
+def apple_deployment_target_flag_name(_os):
+    """compiler flag name which controls deployment target"""
+    return {'Macos': '-mmacosx-version-min',
+            'iOS': '-mios-version-min',
+            'watchOS': '-mwatchos-version-min',
+            'tvOS': '-mappletvos-version-min'}.get(str(_os))
+
+
+class XCRun:
```

Comment:
Please, elaborate a use case.

Reply:
for instance, cross-comping to iOS requires `-isysroot` option to be passed
so you could use it like:
```
sdk_name = apple_sdk_name(self.settings) # e.g. iphoneos
xcrun = XCRun(sdk_name)
CFLAGS = '--isysroot="%s"' % xcrun. sdk_path
```